src/main/java/com/shortify/urlshortener/model/ShortenedUrl.java

```diff
@@ -1,4 +1,29 @@
 package com.shortify.urlshortener.model;
 
-public record ShortenedUrl() {
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+
+import java.util.UUID;
+
+@Entity
+public record ShortenedUrl(
+        @Id
+        @GeneratedValue(strategy = GenerationType.IDENTITY)
+        Long id,
+        String longURL,
+        String shortURLId
+) {
+        public ShortenedUrl{
+                if (shortURLId() == null){
+                        shortURLId = UUID.randomUUID().toString();
```

Comment:
```suggestion
                        shortURLId = UUID.randomUUID().toString();
```
Delegar la generación del UUID desde el Service.

Reply:
Listo.